src/dodal/devices/oav/oav_to_redis_forwarder.py

```diff
@@ -95,7 +110,8 @@ async def _get_frame_and_put_to_redis(self, response: ClientResponse):
     async def _open_connection_and_do_function(
         self, function_to_do: Callable[[ClientResponse, str | None], Awaitable]
     ):
-        stream_url = await self.stream_url.get_value()
+        source = await self.selected_source.get_value()
```

Comment:
Could: use `asyncio.gather` for consecutive reads

Reply:
I don't think I can here because the read of `stream_url` needs the value from the read of `source`. `asyncio.gather` will try and read them in parallel